engine/classes/Elgg/EmailService.php

```diff
@@ -142,5 +150,185 @@ public function transport(Email $email) {
 
 		return true;
 	}
+	
+	/**
+	 * Build the body part of the e-mail message
+	 *
+	 * @param Email $email Email
+	 *
+	 * @return \Zend\Mime\Message
+	 */
+	protected function buildMessageBody(Email $email) {
```

Comment:
Why not just attach files directly to Email instance?

Reply:
could be, but than we'd have to maybe rewrite a lot of our current code to get it into the right place.

Now it works with all the current code in place, just add `attachments` to the `$params` array and it works

Reply:
I just untangled 10 years of email patching. Save your future self some time and do it right from the start with all the tests in place.